lib/cretonne/meta/semantics/smtlib.py

```diff
@@ -0,0 +1,150 @@
+"""
+Tools to emit SMTLIB bitvector queries encoding concrete RTLs containing only
+primitive instructions.
```

Comment:
This is ok, but depending on the z3 Python bindings would also be fine. We just don't want z3 to be a build dependency, but it can be a proof dependency.